db/DB.js

```diff
@@ -101,18 +101,40 @@ const DB = {
     });
   },
 
-  async addSubscriber(email) {
+  /**
+   * Add a subscriber:
+   * 1. Add a record to subscribers
+   * 2. Immediately call _verifySubscriber
+   * 3. For FxA subscriber, add refresh token and profile data
+   *
+   * @param {string} email to add
+   * @param {string} fxaRefreshToken from Firefox Account Oauth
+   * @param {string} fxaProfileData from Firefox Account
+   * @returns {object} subscriber knex object added to DB
+   */
+  async addSubscriber(email, fxaRefreshToken=null, fxaProfileData=null) {
     const emailHash = await this._addEmailHash(getSha1(email), email, true);
-    const verifiedSubscriber = await this._verifySubscriber(emailHash);
-    return verifiedSubscriber[0];
+    const verified = await this._verifySubscriber(emailHash);
+    const verifiedSubscriber = verified[0];
```

Comment:
I realize this is the same as the old code behavior, but should we check that `verified[0]` (or `verifiedSubscriber`) is non-null before using it?

Reply:
Good point. I'll add a check, and I'll actually add a low-level `try-catch` block around the subscription to basket since we may get errors beyond our control there, and we don't need/want to show those low-level errors to the user.